node/lib/task.ts

```diff
@@ -511,7 +511,11 @@ export function stats(path: string): FsStats {
  * @returns   boolean 
  */
 export function exist(path: string): boolean {
-    return path && fs.existsSync(path);
+    try {
+        return path && fs.statSync(path) != null;
+    } catch (err) {
+        return false;
```

Comment:
We can't do this.  You are conclusively saying it doesn't exist if it throws.  For example, if you don't have permissions to the file, then you will eat that and return conclusively that it doesn't exist.  That's not true.

Instead the task should just use exist inline.  That task should be wrapped in one outer try/catch (see ShellScript task).  If it fails to check if a file exists, it should fail.  And it will fail with a clear message EPERM in my example instead of hiding the issue leading to a harder to diagnose issue downstream when the task proceeds to do work on the premise that it doesn't exist.